server/src/main/java/org/uiautomation/ios/utils/SimulatorSettings.java

```diff
@@ -285,47 +286,92 @@ public void resetContentAndSettings() {
+        tryToEraseSimulator();
+      }
+    }
+  }
 
-        // Run command 'xcrun simctl shutdown <uuid>'
-        simctlCmd.executeAndWait(false);
+  private void tryToEraseSimulator() {
+    int numTries = 0;
+    boolean successfulReset = false;
+    while ((!successfulReset) && (numTries < NUMBER_TRIES_RESET_DEVICE)) {
+      log.info("Wait " + SLEEP_TIME_BETWEEN_RESET_TRIES + " milliseconds before reattempt.");
+      try {
+        Thread.sleep(SLEEP_TIME_BETWEEN_RESET_TRIES);
+      } catch (InterruptedException e) {
+        // ignored
+      }
+      successfulReset = eraseSimulator();
+      numTries += 1;
+    }
+    if (!successfulReset) {
+      int totalWaitTime = SLEEP_TIME_BETWEEN_RESET_TRIES * NUMBER_TRIES_RESET_DEVICE;
+      String message = "Erase contents and settings still failed. Total waiting time: "
+              + totalWaitTime
+              + ". Now try to shutdown device: " + deviceUUID;
+      log.warning(message);
 
-        // Retry
-        eraseSimulator();
+      Boolean successfulShutdown = shutdownDevice();
+      if (successfulShutdown) {
+        successfulReset = eraseSimulator();
       }
+
+      if (!successfulReset) {
+        message = "Unable to erase contents and settings of this device: " + deviceUUID;
+        log.warning(message);
+        // add more information to exception message
+        message += ". Tried " + NUMBER_TRIES_RESET_DEVICE + " times with with a waiting period of " + SLEEP_TIME_BETWEEN_RESET_TRIES
+                + " millisecond between each attempt. ";
+        if (!successfulShutdown) {
+          message += "Also unable to shut down the device.";
+        }
+        throw new WebDriverException(message);
+      }
+    }
+  }
+
+  private boolean shutdownDevice() {
+    List<String> simctlArgs;
+    simctlArgs = Arrays.asList("xcrun", "simctl", "shutdown", deviceUUID);
+    Command simctlCmd = new Command(simctlArgs, true);
+    int exitCode = simctlCmd.executeAndWait(true);
+
+    // check for success codes
+    if (exitCode == 146) {
+      log.info("This device shuts down already: " + deviceUUID);
+      return true;
+    } else if (exitCode == 0) {
+      log.info("Successfully shut down the device: " + deviceUUID);
+      return true;
     }
+
+    log.warning("Failed to shut down the device: " + deviceUUID
+            + ". Command was used: " + simctlCmd.commandString());
+    return false;
   }
 
   private boolean eraseSimulator() {
     assert instrumentsVersion.getMajor() >= 6;
 
     // Starting with Xcode 6 and later, we can use simctl to do the hard work for us.
-    List<String> simctlArgs = new ArrayList<>();
-    simctlArgs.add("xcrun");
-    simctlArgs.add("simctl");
-    simctlArgs.add("erase");
-    simctlArgs.add(deviceUUID);
-    Command simctlCmd = new Command(simctlArgs, true);
+    List<String> simctlArgs;
+    simctlArgs = Arrays.asList("xcrun", "simctl", "erase", deviceUUID);
+    Command simctlCmd = new Command(simctlArgs, false);
```

Comment:
I changed it because I've already logged its information. With me, the error message getting from `xcrun simclt erase <deviceID> `seems to be lack of information when we are only use them in the log file. It's my opinion. If it should also be logged, I will change `logToConsole` back to `true`